src/nvim/popupmenu.c

```diff
@@ -799,14 +799,25 @@ static void pum_preview_set_text(buf_T *buf, char *info, linenr_T *lnum, int *ma
   Error err = ERROR_INIT;
   Arena arena = ARENA_EMPTY;
   Array replacement = ARRAY_DICT_INIT;
-  char *token = NULL;
-  char *line = os_strtok(info, "\n", &token);
   buf->b_p_ma = true;
-  while (line != NULL) {
-    ADD(replacement, STRING_OBJ(cstr_to_string(line)));
+
+  // Iterate through the string line by line by temporarily replacing newlines with NUL
+  for (char *curr = info, *next; curr; curr = next ? next + 1 : NULL) {
```

Comment:
I don't see where `curr` or `next` could ever be NULL. Thus this is equivalent?:

```suggestion
  for (char *curr = info, *next; true; curr = next + 1) {
```

Reply:
Oh, I see where `next` could be NULL.